app/controllers/groups_controller.rb

```diff
@@ -78,6 +79,27 @@ def undelete
     redirect_to group_path
   end
 
+  def export_students
+    @group = Group.find params.require :id
+    authorize @group
+
+    respond_to do |format|
+      format.html
+      format.csv do
+        filename = ["#{@group.group_name} - Students", Time.zone.today.to_s].join(' ')
+        send_data csv_from_array_of_hashes(get_student_export_data(@group.students)), filename:, content_type: 'text/csv'
+      end
+    end
+  end
+
+  def get_student_export_data(students)
+    students.map do |student|
```

Comment:
This logic should be in the student model. Controller should only deal with connecting routes, models and views.